File: back-end/src/models/todo.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Deserialize, utoipa::ToSchema)]
pub struct CreateTodoRequest {
    pub title: String,
    pub status: Option<String>,
}

#[derive(Debug, Deserialize, utoipa::ToSchema)]
pub struct UpdateTodoRequest {
    pub title: Option<String>,
    pub completed: Option<bool>,
    pub status: Option<String>,
    pub position: Option<i32>,
}

#[derive(Debug, Deserialize, utoipa::ToSchema)]
pub struct ReorderTodoItem {
    pub id: Uuid,
    pub status: String,
    pub position: i32,
}

#[derive(Debug, Deserialize, utoipa::ToSchema)]
pub struct ReorderTodosRequest {
    pub items: Vec<ReorderTodoItem>,
}

#[derive(Debug, Serialize, FromRow, utoipa::ToSchema)]
pub struct TodoResponse {
    pub id: Uuid,
    pub user_id: Uuid,
    pub title: String,
    pub completed: bool,
    pub status: String,
    pub position: i32,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

File: bun_hono/src/models/todo.ts
export interface CreateTodoRequest {
  title: string;
  status?: string | null;
  assignee_id?: string | null;
}

export interface UpdateTodoRequest {
  title?: string | null;
  completed?: boolean | null;
  status?: string | null;
  position?: number | null;
  assignee_id?: string | null;
}

export interface ReorderTodoItem {
  id: string;
  status: string;
  position: number;
}

export interface ReorderTodosRequest {
  items: ReorderTodoItem[];
}

export interface TodoResponse {
  id: string;
  reporter_id: string;
  reporter_email: string;
  assignee_id: string | null;
  assignee_email: string | null;
  title: string;
  completed: boolean;
  status: string;
  position: number;
  created_at: string;
  updated_at: string;
}
